munch/__init__.py

```diff
@@ -259,6 +263,38 @@ def __repr__(self):
             type(self).__name__, self.__undefined__, dict.__repr__(self))
 
 
+class DefaultFactoryMunch(defaultdict, Munch):
+    """ A Munch that calls a user-specified function to generate values for
+        missing keys like collections.defaultdict.
+
+        >>> b = DefaultFactoryMunch(list, {'hello': 'world!'})
+        >>> b.hello
+        'world!'
+        >>> b.foo
+        []
+        >>> b.bar.append('hello')
+        >>> b.bar
+        ['hello']
+    """
+
+    def __init__(self, default_factory, *args, **kwargs):
```

Comment:
Doing it this way to force the user to pass in a default_factory since defaultdict doesn't require it.